network-mux/src/Network/Mux.hs

```diff
@@ -136,13 +145,55 @@ muxStart tracer peerid app bearer = do
                         | ptcl <- [minBound..maxBound]
                         , mode <- [ModeInitiator, ModeResponder] ]
 
+    -- Construct an array used for controlling when an initiator application should start and
+    -- a return value for the initiator applications
+    setupInitiatorCtrlTbl :: m (MiniProtocolInitiatorControlTable ptcl m a)
+    setupInitiatorCtrlTbl = MiniProtocolInitiatorControlTable
+             . array (minBound, maxBound)
+           <$> sequence [ do q <- atomically newEmptyTMVar
+                             p <- atomically newEmptyTMVar
+                             return (ptcl, (q, p))
+                        | ptcl <- [minBound..maxBound]
+                        ]
+
+    -- Construct an array used for providing the return value for responder applications
+    setupResponderCtrlTbl :: m (MiniProtocolResponderControlTable ptcl m b)
+    setupResponderCtrlTbl = MiniProtocolResponderControlTable
+             . array (minBound, maxBound)
+           <$> sequence [ do q <- atomically newEmptyTMVar
+                             return (ptcl, q)
```

Comment:
why not `(ptcl, ) <$> newEmptyTMVarM`?

Reply:
I think one can rewrite `... <$> sequence ...` with `traverse`

Reply:
Its is written in the same style as `setupTbl`, and I don't see a reason to change it.

Reply:
> why not `(ptcl, ) <$> newEmptyTMVarM`?

Will address.